packages/protocol/contracts/L1/v1/V1Proposing.sol

```diff
@@ -112,6 +121,29 @@ library V1Proposing {
         emit BlockProposed(s.nextBlockId++, meta);
     }
 
+    function whitelistProposer(
+        LibData.State storage s,
+        address proposer,
+        bool enabled
+    ) public {
+        assert(s.whitelistProposers);
+        require(
+            proposer != address(0) && s.proposers[proposer] != enabled,
+            "L1:precondition"
```

Comment:
since it's temporary it is probably fine but I am not usually a fan of these vague require messages. "L1: proposer not whitelisted" is more clear.

Reply:
Short message is fine as long as it's unique in each transaction for locating the error.